auto_nag/tests/test_utils.py

```diff
@@ -37,3 +41,54 @@ def test_hide_personal_info(self):
             "include_fields=_default,attachments&changed_field_to=" + \
             "RESOLVED&api_key=xyzxyzxyz&resolution=FIXED"
         assert "xyzxyzxyz" not in hide_personal_info(sample_exception)
+
+    def test_urljoin(self):
+        api_root = 'https://bugzilla.mozilla.org/bzapi/'
+        url = urljoin(
+            api_root, 'bug/%s/attachment?%s' %
+            (1233, '&changed_before=' +
+                '2010-12-26&product=Core,Firefox&changed_field=status&' +
+                'changed_after=2010-12-24&include_fields=_default,' +
+                'attachments&changed_field_to=RESOLVED&resolution=FIXED'))
+
+        expected_url = 'https://bugzilla.mozilla.org/bzapi/bug/1233/' + \
+            'attachment?&changed_before=2010-12-26&product=Core,Firefox&' + \
+            'changed_field=status&changed_after=2010-12-24&include_fields=' + \
+            '_default,attachments&changed_field_to=RESOLVED&resolution=FIXED'
+        assert url == expected_url
+
+    def test_qs(self):
+        param = {'product': ['Core,Firefox']}
+        _qs = qs(**param)
+        expected_qs = '&product=Core,Firefox'
+        assert _qs == expected_qs
+
+    def test_get_credentials(self):
+        # if already bztoolsrc is present, backup the existing '~/.bztoolsrc'
+        expand_usr = os.path.expanduser('~')
+        rcfile = os.path.expanduser('~/.bztoolsrc')
+        if os.path.exists(rcfile):
+            cmd = 'mv ' + rcfile + ' ' + expand_usr + '/.bztoolsrc_backup'
+            os.system(cmd)
+        cmd = 'python ' + get_project_root_path() + \
+            'auto_nag/tests/_get_credential.py'
+        child = pexpect.spawn(cmd, timeout=180)
+        child.expect('username')
+        child.sendline('test_username')
+        child.expect('password')
+        child.sendline('test_password')
+        child.expect(pexpect.EOF)
+        child.close()
+
+        credentials = get_credentials()
+
+        # Removing the ~/.bztoolsrc which generated for testing.
+        cmd = 'rm ' + rcfile
+        os.system(cmd)
+
+        # copy back bztoolsrc_backup to ~/.bztoolsrc
+        rcfile_backup = os.path.expanduser('~/.bztoolsrc_backup')
+        if os.path.exists(rcfile_backup):
+            cmd = 'mv ' + rcfile_backup + ' ' + expand_usr + '/.bztoolsrc'
```

Comment:
https://docs.python.org/2/library/shutil.html
please use shutil.move

I won't acces any PR containing call to system command when alternatives are available in the language.

On the global level, doing this is a bad programming habit because:
* mv could be aliased on the system and have a diff system (alias mv="mv -i" for example)
* mv might not exists on the system (Windows?)
* if it fails (wrong permissions), it is going to be hard to catch the correct exception and manage the error.